### ISSUE TITLE
Spark memory storage in RStudio Server

### ISSUE THREAD

**Issue by hiltonmbr** [rstudio/sparklyr#275]
I used the following procedures to increase the storage memory for RDD partitions:

library(sparklyr)
dir <- paste0(spark_home_dir(),"/R/lib/")
library("SparkR",lib.loc =dir)
sc <- sparkR.session(master = "local[*]","SparkR",
                     sparkConfig = list(spark.executor.memory="4g",
                                        spark.driver.memory="4g"))

Launching java with spark-submit command /Users/Hilton_iMac/Library/Caches/spark/spark-2.0.1-bin-hadoop2.7/bin/spark-submit   --driver-memory "4g" sparkr-shell /var/folders/hw/51jrzlkx0vgdrnv_jsz_jcnh0000gn/T//RtmpzwBjzM/backend_port9da71291146 

Taking a look on spark UI: http://localhost:4040/executors/
![captura de tela 2016-10-16 as 10 24 35](https://cloud.githubusercontent.com/assets/1009086/19417731/d24b1bf2-938a-11e6-8427-a7c45deb3a90.png)

On the other hand, doing the same with sparklyr in RStudio Server I cannot change the storage memory for RDD partitions:

library(sparklyr)
sc <- spark_connect(master = "local[*]",
                    config = list("sparklyr.shell.executor-memory"= "4g",
                                  "sparklyr.shell.driver-memory"="4g"))
spark_web(sc)

INFO MemoryStore: MemoryStore started with capacity 366.3 MB
INFO NettyBlockTransferService: Server created on 127.0.0.1:52068
INFO BlockManagerMasterEndpoint: Registering block manager 127.0.0.1:52068 with 366.3 MB RAM
 WARN SparkContext: Use an existing SparkContext, some configuration may not take effect.

It does not work:
![captura de tela 2016-10-16 as 10 30 53](https://cloud.githubusercontent.com/assets/1009086/19417765/a834cfec-938b-11e6-867e-f9f87d286da7.png)

**Comment by kykuo** [comment]
@hiltonmbr see https://github.com/rstudio/sparklyr/issues/286

**Comment by javierluraschi** [comment]
See http://stackoverflow.com/questions/26562033/how-to-set-apache-spark-executor-memory

You would need to run instead:

```{r}
config <- spark_config()
config$`sparklyr.shell.driver-memory` <- "4G"
sc <- spark_connect(master = "local", config = config)
```

Reason is that, in local mode, the setting needs to be set before starting the `JVM`.

**Comment by javierluraschi** [comment]
@hiltonmbr is it possible that the `SparkR` application is still running and taking all the remaining resources?

I took a look at #286 and verified this would have worked in my environment: `config = list("sparklyr.shell.executor-memory"= "4g", "sparklyr.shell.driver-memory"="4g"))
`

Could you share the `spark_log(sc)` after `spark_connect`? We could double check that memory parameters are being passed correctly to `spark_submit`.

Also, mind sharing `sessionInfo()`?

**Comment by hiltonmbr** [comment]
@javierluraschi after update sparklyr the memory allocation is working fine:
library(sparklyr)
 sc <- spark_connect(master = "local[*]",
                     config = list("sparklyr.shell.executor-memory"= "8g",
                                   "sparklyr.shell.driver-memory"="8g"))
 spark_log(sc)
16/10/30 09:04:10 INFO sparklyr session (6703) starting
16/10/30 09:04:11 INFO sparklyr gateway received command identifier (0)
16/10/30 09:04:11 INFO sparklyr gateway found current session (6703)
16/10/30 09:04:12 INFO sparklyr gateway created backend for session (6703)
16/10/30 09:04:12 INFO sparklyr gateway waiting for r process to end in session (6703)
16/10/30 09:04:13 INFO SparkContext: Running Spark version 2.0.1
16/10/30 09:04:14 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
16/10/30 09:04:14 WARN Utils: Your hostname, lema resolves to a loopback address: 127.0.1.1; using 192.168.0.100 instead (on interface enp2s0)
16/10/30 09:04:14 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
16/10/30 09:04:14 INFO SecurityManager: Changing view acls to: hilton
16/10/30 09:04:14 INFO SecurityManager: Changing modify acls to: hilton
16/10/30 09:04:14 INFO SecurityManager: Changing view acls groups to: 
16/10/30 09:04:14 INFO SecurityManager: Changing modify acls groups to: 
16/10/30 09:04:14 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(hilton); groups with view permissions: Set(); users  with modify permissions: Set(hilton); groups with modify permissions: Set()
16/10/30 09:04:14 INFO Utils: Successfully started service 'sparkDriver' on port 32891.
16/10/30 09:04:14 INFO SparkEnv: Registering MapOutputTracker
16/10/30 09:04:15 INFO SparkEnv: Registering BlockManagerMaster
16/10/30 09:04:15 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-e9ec6e90-5d2e-42d6-8089-27345a9aafa2
16/10/30 09:04:15 INFO MemoryStore: MemoryStore started with capacity 4.1 GB
16/10/30 09:04:15 INFO SparkEnv: Registering OutputCommitCoordinator
16/10/30 09:04:15 INFO Utils: Successfully started service 'SparkUI' on port 4040.
16/10/30 09:04:15 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://192.168.0.100:4040
16/10/30 09:04:15 INFO SparkContext: Added JAR file:/usr/local/lib/R/site-library/sparklyr/java/sparklyr-2.0-2.11.jar at spark://192.168.0.100:32891/jars/sparklyr-2.0-2.11.jar with timestamp 1477829055671
16/10/30 09:04:15 INFO Executor: Starting executor ID driver on host localhost
16/10/30 09:04:15 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 45378.
16/10/30 09:04:15 INFO NettyBlockTransferService: Server created on 192.168.0.100:45378
16/10/30 09:04:15 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 192.168.0.100, 45378)
16/10/30 09:04:15 INFO BlockManagerMasterEndpoint: Registering block manager 192.168.0.100:45378 with 4.1 GB RAM, BlockManagerId(driver, 192.168.0.100, 45378)
16/10/30 09:04:15 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 192.168.0.100, 45378)
16/10/30 09:04:16 WARN SparkContext: Use an existing SparkContext, some configuration may not take effect.
16/10/30 09:04:16 WARN SparkContext: Use an existing SparkContext, some configuration may not take effect.
16/10/30 09:04:16 INFO HiveSharedState: Warehouse path is '/home/Volumes/LEMA/DataSA/IBGE/Estimativas_Populacao/spark-warehouse'.

**Comment by javierluraschi** [comment]
Thanks for the update, closing this issue then.
